tests/countries/test_cocos_islands.py

```diff
@@ -0,0 +1,281 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.cocos_islands import CocosIslands, CC, CCK
+from tests.common import CommonCountryTests
+
+
+class TestCocosIslands(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1985, 2050)
+        super().setUpClass(CocosIslands, years=years, years_non_observed=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(CocosIslands, CC, CCK)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(CocosIslands(years=1984))
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1985, 2050)))
+        obs_dt = (
+            "2011-01-03",
+            "2012-01-02",
+            "2017-01-02",
+            "2022-01-03",
+            "2023-01-02",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_australia_day(self):
+        name = "Australia Day"
+        self.assertHolidayName(name, (f"{year}-01-26" for year in range(1985, 2050)))
+        obs_dt = (
```

Comment:
```suggestion
        obs_dt = (
            "2013-01-28",
            "2014-01-27",
            "2019-01-28",
```
For all holidays, need 4-5 dates.

Reply:
_All_ holidays that have observed feature.